react/src/lib/components/DeckGLMap/layers/terrain/map3DLayer.ts

```diff
@@ -133,6 +134,10 @@ async function load_mesh_and_texture(
         console.log("Error. One or both of texture and mesh must be given!");
     }
 
+    // Serve the worker as a local url. See: https://deck.gl/docs/developer-guide/loading-data#loaders-and-web-workers
+    const blob = new Blob([WORKER_SCRIPT], { type: "text/plain" });
+    const url = window.URL.createObjectURL(blob);
```

Comment:
This would require `script-src blob:` in CSP, which is equivalent to allowing `unsafe-eval`, as confirmed by https://www.w3.org/TR/CSP2/#source-list-guid-matching:
> Especially for the [default-src](https://www.w3.org/TR/CSP2/#default_src) and [script-src](https://www.w3.org/TR/CSP2/#script_src) directives, policy authors should be aware that allowing "data:" URLs is equivalent to unsafe-inline and allowing "blob:" or "filesystem:" URLs is equivalent to unsafe-eval.

For deployed applications that want tight security settings, we could perhaps offer the component consumer to provide a `url` where the web worker script is hosted, and if not provided we do as done here (create a `blob` url` for `script` usage)?